tests/index.js

```diff
@@ -75,26 +122,68 @@ describe('core/block processor', function () {
       (async () => {
 
         try {
-          await channel.assertQueue(`app_${config.rabbit.serviceName}_test.transaction`);
+          let queue1 = await channel.assertQueue(`app_${config.rabbit.serviceName}_test.transaction`);
           await channel.bindQueue(`app_${config.rabbit.serviceName}_test.transaction`, 'events', `${config.rabbit.serviceName}_transaction.*`);
+          expect(queue1.messageCount).to.equal(2);
         } catch (e) {
           channel = await amqpInstance.createChannel();
         }
 
-        return await new Promise(res => {
-          channel.consume(`app_${config.rabbit.serviceName}_test.transaction`, res, {noAck: true})
-        })
+        getAmqpMessage = channel.consume.bind(channel, `app_${config.rabbit.serviceName}_test.transaction`);
+        getAndCheckAmqpMessage(getAmqpMessage);
+        getAndCheckAmqpMessage(getAmqpMessage);
+      
       })(),
       (async () => {
         let ws = new WebSocket('ws://localhost:15674/ws');
         let client = Stomp.over(ws, {heartbeat: false, debug: false});
         return await new Promise(res =>
           client.connect('guest', 'guest', () => {
-            client.subscribe(`/exchange/events/${config.rabbit.serviceName}_transaction.*`, res)
+            getWsMessage = client.subscribe.bind(client, `/exchange/events/${config.rabbit.serviceName}_transaction.*`);
+            getAndCheckWsMessage(getWsMessage)
+            getAndCheckWsMessage(getWsMessage)
```

Comment:
why are you calling this function twice?

Reply:
I check that two message i recieved - one from user with from field and two with to field